.changeset/metal-pumas-walk.md

```diff
@@ -0,0 +1,43 @@
+---
+'@astrojs/node': major
+---
+
+# Standalone mode for the Node.js adapter
+
+New in `@astrojs/node` is support for __standalone mode__. With standalone mode you can start your production server without needing to write any server JavaScript yourself. The server starts simply by running the script like so:
+
+```shell
+node ./dist/server/entry.mjs
+```
+
+To enable standalone mode set the new `mode` to `'standalone'` option in your Astro config:
+
+```js
+import { defineConfig } from 'astro/config';
+import nodejs from '@astrojs/node';
+
+export default defineConfig({
+  output: 'server',
+  adapter: nodejs({
+    mode: 'standalone'
+  })
+});
+```
+
+See the @astrojs/node documentation to learn all of the options available in standalone mode.
+
+## Breaking change
```

Comment:
thinking more about this, if you default to `middleware` and make this option required, do you still need the major version?

Reply:
It wouldn't be required if it had a default, no?

We could default to `middleware`. My thought was that these two modes are pretty fairly different and it didn't feel right to "prefer" one over the other. Kind of like Vercel serverless vs edge, just completely different things.

As far as semver, given that this is an integration I didn't think it mattered as much to do a semver major change.

But I don't feel too strongly on either of these things.

Reply:
yea, agreed that because its an integration this doesn't matter as much / I'm okay with the plan for a major. And I agree it makes more sense as required, or as defaulting to standalone. Defaulting to `middleware` is probably the least natural of the 3...

Reply:
Ok, I think we should do a major in that case and require the option. We can track usage in the future and if `standalone` is overwhelming (and once it becomes more stable) we can make it the default if we want to.